app/scripts/services/DateRangeService.js

```diff
@@ -0,0 +1,39 @@
+import moment from 'moment';
+
+angular.module('confRegistrationWebApp')
+  .service('DateRangeService', function DateChangeEmitter() {
+    var subscribed = {};
+    var cache = {};
+
+    this.subscribe = function (emitter, callback) {
+      if (angular.isUndefined(subscribed[emitter])) subscribed[emitter] = [];
+
+      subscribed[emitter].push(callback);
+
+      if (angular.isDefined(cache[emitter])) {
+        callback(cache[emitter]);
+      }
+    };
+
+    this.emitChange = function (emitter, value) {
+      cache[emitter] = value;
+
+      angular.forEach(subscribed[emitter], function(callback) {
+        callback(value);
+      });
+    };
+
+    this.calculateDateRange = function (startDate, endDate) {
+      if (angular.isUndefined(startDate) || startDate === null || startDate === '') {
+        return 1;
+      }
+      if (angular.isUndefined(endDate) || endDate === null || endDate === '') {
```

Comment:
I believe that this `angular.isUndefined(endDate) || endDate === null || endDate === ''` can be changed to be `!endDate`.